source/chip/STM32/STM32L0/Kconfig-chipOptions

```diff
@@ -0,0 +1,1597 @@
+	default CHIP_STM32L0_RCC_SYSCLK_HSE if CHIP_STM32L0_RCC_HSE_ENABLE
+	default CHIP_STM32L0_RCC_SYSCLK_HSI16
+	help
+		Select system clock source.
+
+config CHIP_STM32L0_RCC_SYSCLK_HSI16
+	bool "HSI16"
+	help
+		HSI16 oscillator selected as system clock.
+
+config CHIP_STM32L0_RCC_SYSCLK_HSE
+	bool "HSE"
+	depends on CHIP_STM32L0_RCC_HSE_ENABLE
+	help
+		HSE oscillator selected as system clock.
+
+config CHIP_STM32L0_RCC_SYSCLK_PLL
+	bool "PLL"
+	depends on CHIP_STM32L0_RCC_PLL_ENABLE
+	help
+		Main PLL selected as system clock.
+
+config CHIP_STM32L0_RCC_SYSCLK_MSI
+	bool "MSI"
+	depends on CHIP_STM32L0_RCC_MSI_ENABLE
+	help
+		MSI oscillator selected as system clock.
+
+endchoice
+
+endif	# CHIP_STM32L0_STANDARD_CLOCK_CONFIGURATION_ENABLE
+
+if !CHIP_STM32L0_STANDARD_CLOCK_CONFIGURATION_ENABLE
+
+config CHIP_STM32L0_RCC_SYSCLK_FREQUENCY
+	int "Frequency of SYSCLK, Hz"
+	range 1 32000000
+	default 8000000
+	help
+		Frequency of SYSCLK, Hz. RCC must be configured by user to achieve that
+		frequency.
+
+endif	# !CHIP_STM32L0_STANDARD_CLOCK_CONFIGURATION_ENABLE
+
+choice
+	prompt "AHB divider"
+	default CHIP_STM32L0_RCC_AHB_DIV1
+	help
+		AHB clock division factor
+
+		AHBclk = SYSclk / AHBdivider
+
+config CHIP_STM32L0_RCC_AHB_DIV1
+	bool "1"
+	help
+		System clock not divided
+
+config CHIP_STM32L0_RCC_AHB_DIV2
+	bool "2"
+	help
+		System clock divided by 2
+
+config CHIP_STM32L0_RCC_AHB_DIV4
+	bool "4"
+	help
+		System clock divided by 4
+
+config CHIP_STM32L0_RCC_AHB_DIV8
+	bool "8"
+	help
+		System clock divided by 8
+
+config CHIP_STM32L0_RCC_AHB_DIV16
+	bool "16"
+	help
+		System clock divided by 16
+
+config CHIP_STM32L0_RCC_AHB_DIV64
+	bool "64"
+	help
+		System clock divided by 64
+
+config CHIP_STM32L0_RCC_AHB_DIV128
+	bool "128"
+	help
+		System clock divided by 128
+
+config CHIP_STM32L0_RCC_AHB_DIV256
+	bool "256"
+	help
+		System clock divided by 256
+
+config CHIP_STM32L0_RCC_AHB_DIV512
+	bool "512"
+	help
+		System clock divided by 512
+
+endchoice
+
+choice
+	prompt "APB1 (low speed) divider"
+	default CHIP_STM32L0_RCC_APB1_DIV2
+	help
+		APB1 (low speed) clock division factor. APB1 clock must not exceed:
+		- 42MHz for STM32L001, STM32L005, STM32L007, STM32L015 and STM32L017
+		chips,
+		- 50MHz for STM32L010, STM32L011 and STM32L012 chips,
+		- 45MHz for other chips.
+
+		APB1clk = AHBclk / APB1divider
+
+config CHIP_STM32L0_RCC_APB1_DIV1
+	bool "1"
+	help
+		AHB clock not divided
+
+config CHIP_STM32L0_RCC_APB1_DIV2
+	bool "2"
+	help
+		AHB clock divided by 2
+
+config CHIP_STM32L0_RCC_APB1_DIV4
+	bool "4"
+	help
+		AHB clock divided by 4
+
+config CHIP_STM32L0_RCC_APB1_DIV8
+	bool "8"
+	help
+		AHB clock divided by 8
+
+config CHIP_STM32L0_RCC_APB1_DIV16
+	bool "16"
+	help
+		AHB clock divided by 16
+
+endchoice
+
+choice
+	prompt "APB2 (high speed) divider"
+	default CHIP_STM32L0_RCC_APB2_DIV2
+	help
+		APB2 (high speed) clock division factor. APB2 clock must not exceed:
+		- 32MHz
```

Comment:
So it will be sufficient to just leave help description to that one (for APB1 and APB2)?

```
choice
	prompt "APB1 (high speed) divider"
	default CHIP_STM32L0_RCC_APB1_DIV1
	help
		APB1 (high speed) clock division factor.
		
		APB1clk = AHBclk / APB1divider
```

```
choice
	prompt "APB2 (high speed) divider"
	default CHIP_STM32L0_RCC_APB2_DIV1
	help
		APB2 (high speed) clock division factor.
		
		APB2clk = AHBclk / APB2divider
```
Because there is just divider (stm32l0 doesn't have multiplier) of AHBclk and AHBclk cannot exceed 32 MHz.